File: leetcode/leetcode1331.py
#https://leetcode.cn/problems/rank-transform-of-an-array/description/?envType=daily-question&envId=2026-07-12
from typing import List
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        rank= {num:i+1 for i,num in enumerate(sorted(set(arr)))}
        return [rank[num] for num in arr]

File: leetcode/leetcode1331.cpp
#include<vector>
#include<set>
#include<unordered_map>
#include<algorithm>
using namespace std;
class Solution {
public:
    vector<int> arrayRankTransform(vector<int>& arr) {
        vector<int> arrSorted=arr;
        vector<int> ans=vector<int>(arr.size());
        sort(arrSorted.begin(),arrSorted.end());
        unordered_map<int,int> rank;
        for(auto a:arrSorted){
            if(!rank.count(a)){
                rank[a]=rank.size()+1;
            }
        }
        for(int i=0;i<arr.size();i++){
            ans[i]=rank[arr[i]];
        }
        return ans;
    }
};